clients/fides-js/src/docs/fides-event.ts

```diff
@@ -122,12 +130,30 @@ export interface FidesEvent extends CustomEvent {
       consentMethod?: "accept" | "reject" | "save" | "dismiss" | "gpc";
 
       /**
-       * What toggle (if any) caused this event.
+       * What UI element (if any) triggered this event.
        */
-      servingToggle?: {
-        label: string;
+      trigger?: {
+        /**
+         * The type of element that triggered the event. Additional types may be
+         * added over time, so expect this type to grow.
+         */
+        type: "toggle" | "button" | string;
```

Comment:
I would suggest keeping this strict until additional types are needed and are explicitly added to the type.
```suggestion
        type: "toggle" | "button";
```

Reply:
The reason I did this is because that might be considered a breaking change to code that uses this today. Does that make sense?

Reply:
what code uses it today? Isn't this net new to our system? Regardless it's ok I guess for documentation, but not for the benefit of TS. TS will just allow string now and there's no real reason to include the first 2 stricter checks.

Reply:
OK, agreed! I narrowed this down to the only trigger type currently defined: `"toggle"`.